amazon-kinesis-client/src/test/java/software/amazon/kinesis/common/StreamIdentifierTest.java

```diff
@@ -0,0 +1,94 @@
+        mockStatic(StreamARNUtil.class);
+        when(StreamARNUtil.getStreamARN(eq(streamName), eq(kinesisRegion))).thenReturn(arn);
+        StreamIdentifier actualStreamIdentifier = StreamIdentifier.singleStreamInstance(streamName, kinesisRegion);
+        Assert.assertFalse(actualStreamIdentifier.streamCreationEpochOptional().isPresent());
+        Assert.assertFalse(actualStreamIdentifier.accountIdOptional().isPresent());
+        Assert.assertTrue(actualStreamIdentifier.streamARNOptional().isPresent());
+        Assert.assertEquals(arn, actualStreamIdentifier.streamARNOptional());
+    }
+
+    @Test
+    public void testMultiStreamInstanceWithIdentifierSerialization() {
+        String epoch = "1680616058";
+        Optional<Arn> arn = Optional.ofNullable(Arn.builder().partition("aws").service("kinesis")
+                .accountId(accountId).region(kinesisRegion.toString()).resource("stream/" + streamName).build());
+
+        mockStatic(StreamARNUtil.class);
+        when(StreamARNUtil.getStreamARN(eq(streamName), any(), any())).thenReturn(arn);
+        StreamIdentifier actualStreamIdentifier = StreamIdentifier.multiStreamInstance(
+                Joiner.on(":").join(accountId, streamName, epoch, kinesisRegion));
+        assertActualStreamIdentifierExpected(arn, actualStreamIdentifier);
+    }
+
+    @Test
+    public void testMultiStreamInstanceWithRegionSerialized() {
+        Region serializedRegion = Region.US_GOV_EAST_1;
+        Optional<Arn> arn = Optional.ofNullable(Arn.builder().partition("aws").service("kinesis")
+                .accountId(accountId).region(serializedRegion.toString()).resource("stream/" + streamName).build());
+
+        mockStatic(StreamARNUtil.class);
+        when(StreamARNUtil.getStreamARN(eq(streamName), eq(serializedRegion), any())).thenReturn(arn);
+        StreamIdentifier actualStreamIdentifier = StreamIdentifier.multiStreamInstance(
+                Joiner.on(":").join(accountId, streamName, epoch, serializedRegion), kinesisRegion);
+        assertActualStreamIdentifierExpected(arn, actualStreamIdentifier);
+    }
+
+    @Test
+    public void testMultiStreamInstanceWithoutRegionSerialized() {
+        Optional<Arn> arn = Optional.ofNullable(Arn.builder().partition("aws").service("kinesis")
+                .accountId(accountId).region(kinesisRegion.toString()).resource("stream/" + streamName).build());
+
+        mockStatic(StreamARNUtil.class);
+        when(StreamARNUtil.getStreamARN(eq(streamName), eq(kinesisRegion), any())).thenReturn(arn);
+        StreamIdentifier actualStreamIdentifier = StreamIdentifier.multiStreamInstance(
+                Joiner.on(":").join(accountId, streamName, epoch), kinesisRegion);
+        assertActualStreamIdentifierExpected(arn, actualStreamIdentifier);
+    }
+
+    private void assertActualStreamIdentifierExpected(Optional<Arn> expectedARN, StreamIdentifier actual) {
+        Assert.assertTrue(actual.streamCreationEpochOptional().isPresent());
```

Comment:
How many of these lossy `assertTrue` can you replace w/ stronger expectations? (e.g.,  `assertEquals(Optional.of(EPOCH), actual.sCEO)` ?

Validating _exact_ values yields better tests. These assertions would not flag incorrect results so long as they're non-empty.